daml-assistant/exe/DAML/Assistant.hs

```diff
@@ -17,21 +17,64 @@ import System.FilePath
 import System.Directory
 import System.Process
 import System.Exit
+import System.IO
 import Control.Exception.Safe
 import Data.Maybe
+import Control.Monad.Extra
 
 -- | Run the assistant and exit.
 main :: IO ()
-main = do
+main = displayErrors $ do
     env@Env{..} <- getDamlEnv
     sdkConfigM <- mapM readSdkConfig envSdkPath
     sdkCommandsM <- mapM (fromRightM throwIO . listSdkCommands) sdkConfigM
     userCommand <- getCommand (fromMaybe [] sdkCommandsM)
+
+    whenJust envLatestStableSdkVersion $ \latestVersion -> do
+        let isHead = maybe False isHeadVersion envSdkVersion
+            test1 = not isHead && isJust envProjectPath && envSdkVersion < Just latestVersion
+            test2 = not isHead && not test1 && isJust envDamlAssistantSdkVersion &&
+                fmap unwrapDamlAssistantSdkVersion envDamlAssistantSdkVersion <
+                Just latestVersion
+
+        -- Project SDK version is outdated.
+        when test1 $ do
+            hPutStrLn stderr . unlines $
+                [ "WARNING: Using an outdated version of the DAML SDK in project."
+                , ""
+                , "Please set the sdk-version in the project config daml.yaml"
+                , "to the latest stable version " <> versionToString latestVersion
+                    <> " like this:"
+                , ""
+                , "sdk-version: " <> versionToString latestVersion
+                , ""
+                ]
+
+        -- DAML assistant is outdated.
+        when test2 $ do
```

Comment:
While `test2` is obviously better than `test1` since `2 > 1`, you might want to consider changing this as well :smile: